aptos-move/block-executor/src/view.rs

```diff
@@ -283,9 +311,38 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
         ret
     }
 
+    /// Given a state value, performs deserialization-serialization round-trip
+    /// to replace any aggregator / snapshot values.
+    fn replace_values_with_identifiers(
+        &self,
+        state_value: StateValue,
+        layout: &MoveTypeLayout,
+    ) -> anyhow::Result<StateValue> {
+        state_value.map_bytes(|bytes| {
+            // This call will replace all occurrences of aggregator / snapshot
+            // values with unique identifiers with the same type layout.
+            // The values are stored in aggregators multi-version data structure,
+            // see the actual trait implementation for more details.
+            let patched_value =
+                deserialize_and_replace_values_with_ids(bytes.as_ref(), layout, self).ok_or_else(
+                    || anyhow::anyhow!("Failed to deserialize resource during id replacement"),
+                )?;
+            patched_value
+                .simple_serialize(layout)
+                .ok_or_else(|| {
+                    anyhow::anyhow!(
+                        "Failed to serialize value {} after id replacement",
+                        patched_value
+                    )
+                })
+                .map(|b| b.into())
+        })
+    }
+
     fn get_resource_state_value_impl(
         &self,
         state_key: &T::Key,
+        maybe_layout: Option<&MoveTypeLayout>,
```

Comment:
@gelash  - with rebase, I needed to add maybe_layout back here that you removed in your PR, does this look correct?